scripts/deploy.js

```diff
@@ -10,56 +10,97 @@ const config = {
   region: process.env.DEPLOY_REGION || 'us-west-2',
   uploadDir: 'www',
   index: 'index.html',
-  error: 'index.html', // Setting as the error document allows client-side routing
   enableCloudfront: true
 };
 
 const AWS = require('aws-sdk');
 const s3 = new AWS.S3({ region: config.region });
+const cloudfront = new AWS.CloudFront();
 
 const create = require('s3-website').s3site;
 const deploy = require('s3-website').deploy;
 
-const updateCacheControl = () => {
+function promisify(func, thisContext) {
+  if (thisContext) {
+    func = func.bind(thisContext);
+  }
+  return (...args) => {
+    return new Promise((resolve, reject) => {
+      func(...args, (err, result) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(result);
+        }
+      });
+    })
+  };
+}
+const createPromise = promisify(create);
+const deployPromise = promisify(deploy);
+const getDistributionConfigPromise = promisify(cloudfront.getDistributionConfig, cloudfront);
+const updateDistributionPromise = promisify(cloudfront.updateDistribution, cloudfront);
+const copyObjectPromise = promisify(s3.copyObject, s3);
 
+function updateCacheControl() {
+  console.log('Updating cache settings');
   // NOTE: We need to update the Cache-Control header for the index.html object,
   // as well as any localized index objects (such as es.html, for example),
   // so CloudFront won't cache these objects and they can fetch updated hashed
   // CSS and JS bundles consistently.
-  glob.sync("./www/*.html").forEach((path) => {
+  let updatePromises = glob.sync("./www/*.html").map((path) => {
     splitPath = path.split("/");
     let indexFile = splitPath[splitPath.length - 1];
     let params = {
-      Body: fs.readFileSync("./www/" + indexFile),
+      CopySource: encodeURIComponent(process.env.DEPLOY_DOMAIN + '/' + indexFile),
       Bucket: process.env.DEPLOY_DOMAIN,
       Key: indexFile,
```

Comment:
Does the `Key` parameter also need to be URI encoded, like the `CopySource` parameter is? Here it's an object in the root of the bucket, but it could also be a more complex path.